src/plugins/data_view_editor/public/data_view_editor_service.ts

```diff
@@ -149,6 +135,42 @@ export class DataViewEditorService {
     return dataViewName ? dataViewNames.filter((v) => v !== dataViewName) : dataViewNames;
   };
 
+  private getIndicesMemory: Record<string, Promise<MatchedItem[]>> = {};
+  debounceGetIndices = async (props: { pattern: string; showAllIndices?: boolean | undefined }) => {
+    const debouncedRequest = debounce(this.dataViews.getIndices, 60000, {
```

Comment:
I think I don't understand why we wrap this into `debounce`. since this function would be called only once - I guess there is no need to debounce it. 

Was the intention to make the cache live for 60 seconds? Then I think it doesn't work this way and instead cache is alive for as long as the instance of the service is alive. Because with every `getIndicesMemory` call we get the cached promise from memory instead of calling the debounced function again

Reply:
ah, good catch! No, debounce isn't needed. I've removed it. I think its okay for us to cache the response for the life of the flyout.

Reply:
Ah, Sounds good then 👍 

Should we clean up the cache in case of an error?

Reply:
>Should we clean up the cache in case of an error?

Definitely...